src/main/java/org/springframework/integration/kafka/inbound/KafkaMessageDrivenChannelAdapter.java

```diff
@@ -55,9 +59,16 @@ public void setPayloadDecoder(Decoder<?> payloadDecoder) {
 		this.payloadDecoder = payloadDecoder;
 	}
 
+	public void setAutoCommitOffset(boolean autoCommitOffset) {
+		this.autoCommitOffset = autoCommitOffset;
+	}
+
 	@Override
 	protected void onInit() {
-		this.messageListenerContainer.setMessageListener(new ChannelForwardingMessageListener());
+		this.messageListenerContainer.setAutoCommitOffset(autoCommitOffset);
```

Comment:
Can't we make `KafkaMessageListenerContainer` as smart for that `autoCommitOffset` flag?
I mean determine it by the type of provided `messageListener` instead of the fact to say that end-user is fool :smile: .
And avoid that hard work for the `Assert.isTrue()` in those listenerContainer participants.